packages/snjs/lib/application.ts

```diff
@@ -559,6 +560,28 @@ export class SNApplication {
     return compareVersions(userVersion, ProtocolVersion.V004) >= 0;
   }
 
+  public async getUserSubscription(): Promise<Subscription | undefined> {
+    const response = await this.sessionManager.getSubscription();
+    if (response.error) {
+      throw new Error(response.error.message)
```

Comment:
Hmm why are we throwing? We should propagate this to the client so the user can take action. Otherwise this might block execution of other essential tasks.

Reply:
When throwing an error in an async function it triggers the catch/error of the Promise. That's how it's propagated towards the client.